com.unity.netcode.gameobjects/Tests/Runtime/NetworkTransform/NetworkTransformStateTests.cs

```diff
@@ -59,121 +59,331 @@ public class NetworkTransformStateTests
+            networkTransform.SyncScaleZ = true;
+            networkTransform.InLocalSpace = inLocalSpace;
+            networkTransform.PositionThreshold = positionThreshold;
+            networkTransform.RotAngleThreshold = rotAngleThreshold;
+            networkTransform.ScaleThreshold = scaleThreshold;
+
+            networkTransform.ReplNetworkState.Value = new NetworkTransform.NetworkState
+            {
+                PositionX = initialPosition.x,
+                PositionY = initialPosition.y,
+                PositionZ = initialPosition.z,
+                RotAngleX = initialRotAngles.x,
+                RotAngleY = initialRotAngles.y,
+                RotAngleZ = initialRotAngles.z,
+                ScaleX = initialScale.x,
+                ScaleY = initialScale.y,
+                ScaleZ = initialScale.z,
+                InLocalSpace = inLocalSpace
+            };
+
+            // Step 1: change properties, expect state to be dirty
+            {
+                networkTransform.InLocalSpace = !inLocalSpace;
+                networkTransform.transform.position = new Vector3(3, 4, 5);
+                networkTransform.transform.eulerAngles = new Vector3(30, 45, 90);
+                networkTransform.transform.localScale = new Vector3(1.1f, 0.5f, 2.5f);
+
+                bool isDirty = networkTransform.UpdateNetworkState(networkTransform.ReplNetworkState.Value);
+                networkTransform.ReplNetworkState.SetDirty(isDirty);
+                Assert.IsTrue(isDirty);
+            }
+
+            // Step 2: apply current state locally, expect state to be not dirty/different
+            {
+                networkTransform.ApplyNetworkState(networkTransform.ReplNetworkState.Value);
+
+                bool isDirty = networkTransform.UpdateNetworkState(networkTransform.ReplNetworkState.Value);
+                Assert.IsFalse(isDirty);
+            }
+
+            // Step 3: make changes below and above thresholds
+            {
+                // Position
+                if (!Mathf.Approximately(positionThreshold, 0.0f))
+                {
+                    var position = networkTransform.transform.position;
+
+                    // PositionX
+                    {
+                        position.x += positionThreshold / 2;
+                        networkTransform.transform.position = position;
+                        Assert.IsFalse(networkTransform.UpdateNetworkState(networkTransform.ReplNetworkState.Value));
+
+                        position.x += positionThreshold * 2;
+                        networkTransform.transform.position = position;
+                        Assert.IsTrue(networkTransform.UpdateNetworkState(networkTransform.ReplNetworkState.Value));
+
+                        networkTransform.ApplyNetworkState(networkTransform.ReplNetworkState.Value);
+                        Assert.IsFalse(networkTransform.UpdateNetworkState(networkTransform.ReplNetworkState.Value));
```

Comment:
I should also comment inline why we're expecting false, true, false from these API calls.